crates/bevy_ecs/src/reflect/map_entities.rs

```diff
@@ -73,3 +73,28 @@ impl<C: Component + MapEntities> FromType<C> for ReflectMapEntities {
         }
     }
 }
+
+#[derive(Clone)]
+pub struct ReflectMapEntitiesResource {
```

Comment:
Do we need this type? I imagine `ReflectMapEntities` should be enough, right?

Reply:
Gah! You scared the crap out of me. This comment popped up right in code while I was working.

Ah, no we probably don't. (But I wanted to separate this out while I was figuring out how this worked.)

Reply:
The issue, at the moment, is the C: Component constraint on FromType<C>. I'm not sure how to work around that in the same type.

Reply:
Ah I think I see.

Reply:
> Gah! You scared the crap out of me. This comment popped up right in code while I was working.

Lol I didn't even know that was a thing 😂

> The issue, at the moment, is the C: Component constraint on FromType. I'm not sure how to work around that in the same type.

Oh that's unfortunate. Based on the impl, do you think we'd be able to merge them somehow?

If not, we might be forced to use a new type here (like you have already)

Reply:
I don't expect that merging to be possible: we need the bound in order to be able to extract the resource and modify it.